File: inorder.java
import java.util.*;
import java.io.*;


// Definition for a binary tree node.
 public class TreeNode {
    int val;
     TreeNode left;
     TreeNode right;
     TreeNode() {}
     TreeNode(int val) { this.val = val; }
     TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
     }
 }
class Solution{
    private void dfs(TreeNode node, List<Integer> inorder){
        if(node == null)
           return;
        
        dfs(node.left, inorder);
        inorder.add(node.value);
        dfs(node.right, inorder);
    }

    public List<Integer> inorderTraversal(TreeNode root){
        List<Integer> inorder = new ArrayList<Integer>();
        dfs(root, inorder);
        return inorder;
    }
}

File: inorder.cpp
#include<iostream>
using namespace std;

 
 // Definition for a binary tree node.
struct TreeNode {
      int value; 
      TreeNode *left;
      TreeNode *right;
      TreeNode() : val(0), left(nullptr), right(nullptr) {}
     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
      TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
  };
 
class Solution{
    private :
            void dfs(TreeNode *node, vector<int> &inorder){
                if(node == NULL) return{
                    return;
                }

                dfs(node->left, inorder);
                inorder.push_back(node->value);
                dfs(node->right, inorder);
            }

    public: 
        vector<int> inorderTraversal(TreeNode* root){
            vector<int> inorder;
            dfs(root, inorder);
            return inorder;
        }
};